packages/expo-document-picker/src/ExpoDocumentPicker.web.ts

```diff
@@ -28,31 +28,18 @@ export default {
     document.body.appendChild(input);
 
     return new Promise((resolve, reject) => {
-      input.addEventListener('change', () => {
+      input.addEventListener('change', async () => {
         if (input.files) {
-          const targetFile = input.files[0];
-          const mimeType = targetFile.type;
-          const reader = new FileReader();
-          reader.onerror = () => {
-            reject(new Error(`Failed to read the selected media because the operation failed.`));
-          };
-          reader.onload = ({ target }) => {
-            const uri = (target as any).result;
-            resolve({
-              canceled: false,
-              type: 'success',
-              uri,
-              mimeType,
-              assets: [],
-              name: targetFile.name,
-              file: targetFile,
-              lastModified: targetFile.lastModified,
-              size: targetFile.size,
-              output: input.files,
-            });
-          };
-          // Read in the image file as a binary string.
-          reader.readAsDataURL(targetFile);
+          const results: Promise<DocumentPickerAsset>[] = [];
+          for (let i = 0; i < input.files.length; i++) {
+            results.push(readFile(input.files[i]));
+          }
+          try {
+            const assets = await Promise.all(results);
```

Comment:
since we are trying to resolve multiple results here, i am not sure whether `Promise.all()` is correct here. if there are multiple items in results but just one rejects, it will rejects all the promise. could you check whether it is possible to reject from just one item?

wondering if we'd better to use `Promise.allSettled` here. how do you think?

Reply:
If we fail to read a file on native we'll reject the entire call so this was just to keep the behaviour the same on web.

Reply:
Would returning only some of the selected files end up being confusing? You'd also have to handle some way of explaining that to the user.

Reply:
thanks for sharing the point. that's fair.
only the remaining nit comment about `readAsync` for naming convention. it's good to merge 🚀

Reply:
Thanks @Kudo 👍